### PR TITLE
Physical fes

### PR DESCRIPTION
**Pull request by santiagobadia**
Hi @fverdugo 

Please take a look at this pull request.

I have created a `RefTrait` for `CellField` and `CellBasis` in order to determine whether they are defined in the physical of reference space. 

As a result, I have created some additional internal functions for the physical space case. 

I have created a `FunctionField` that given a function creates a `Field` object, that I have needed in the interpolation subroutines at the physical space.

I have fixed many typos I have bumped into, done some limited renaming, etc.

I have added tests for all the previous machinery, being able to solve problems (e.g., using Lagrangian or Raviart-Thomas) defined *in the physical space*.

There are a couple of thinks that can be done: 1) create new `Kernel`s for performance, e.g., the inverse of the matrix for change of basis, 2) the `to_physical` and `to_ref` functions but I think it is not that important.

Take a look, make comments, etc.

### CODE REVIEW

File: src/Geometry/CellFields.jl [hunk 1]
```diff
@@ -194,6 +207,11 @@ function get_cell_map(cf::GenericCellField)
   cf.cell_map
 end
 
+function RefTrait(::Type{<:GenericCellField{R}}) where {R}
+  Val{R}()
+end
+
+RefTrait(a::GenericCellField) = RefTrait(typeof(a))
```
Comment (fverdugo) [inline src/Geometry/CellFields.jl]: 214 Not needed if you define a default API in `CellFieldLike`

File: src/ReferenceFEs/Dofs.jl [hunk 2]
```diff
@@ -129,12 +129,12 @@ The result is numerically equivalent to
...
 but it is described with a more memory-friendly lazy type.
 """
-function evaluate_dof_array(dof::AbstractArray,field::AbstractArray)
+function evaluate_dof_array(dof::AbstractArray,field::AbstractArray,::Val{true})
   k = DofEval()
   apply(k,dof,field)
 end
 
-function evaluate_dof_array(dof::AbstractArray{<:Dof},field::AbstractArray)
+function evaluate_dof_array(dof::AbstractArray{<:Dof},field::AbstractArray,::Val{false})
```
Comment (fverdugo) [inline src/ReferenceFEs/Dofs.jl]: Same as previous comment.

**Comment by santiagobadia** [comment]
Hi @fverdugo 

I have made the changes, can you take a look?

I have left a comment with @fverdugo asking you what to do in a particular place, very minor.

Bests

--Santi

File: src/FESpaces/CellBases.jl [hunk 3]
```diff
@@ -284,6 +297,9 @@ function operate(op,a,b::CellMatrixField)
   operate(op,_a,b)
 end
 
+RefStyle(::CellMatrixField) = Val{true}()
+# @santiagobadia : @fverdugo can you check what we need here???
```
Comment (santiagobadia) [inline src/FESpaces/CellBases.jl]: @fverdugo can you change it accordingly

**Comment by fverdugo** [comment]
@santiagobadia  I have fixed a number of issues and pushed to this branch. Take a look to the changes if you want.

 Perhaps, before merging it would be a good idea to test the physical dofs for multi-field case. Can you write some test?

**Pull request merged**